File: D_Stack.java
import java.util.*;

public class D_Stack {

    public static void main(String args[]) {
        Scanner sc = new Scanner(System.in);
        Stack<Integer> s = new Stack<>();
        int n = sc.nextInt();
        while (n-- > 0) {
            String str = sc.next();
            if (str.equals("push")) {
                int x = sc.nextInt();
                s.push(x);
            } else if (str.equals("top")) {
                System.out.println(s.peek());
            } else
                s.pop();

        }
    }
}

File: D_Stack.cpp
#include <iostream>
#include <vector>
#include <cmath>
#include <stack>
using namespace std;
int main()
{
    stack<int> myStack;
    int q;
    int v;
    string s;
    cin >> q;

    for (int i = 0; i < q; i++)
    {
        cin >> s;
        if (s == "push")
        {
            cin >> v;
            myStack.push(v);
        }
        else if (s == "pop")
        {
            myStack.pop();
        }
        else if (s == "top")
        {
            cout << myStack.top() << "\n";
        }
    }
    return 0;
}